provider/cluster_autoscaler_resource.go

```diff
@@ -0,0 +1,703 @@
+		Autoscaler().Update().Body(autoscaler).SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed updating cluster autoscaler",
+			fmt.Sprintf(
+				"Failed updating autoscaler for cluster '%s': %v",
+				plan.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	object := update.Body()
+	state := &ClusterAutoscalerState{}
+	populateAutoscalerState(object, plan.Cluster.Value, state)
+
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *ClusterAutoscalerResource) Delete(ctx context.Context, request tfsdk.DeleteResourceRequest,
+	response *tfsdk.DeleteResourceResponse) {
+	state := &ClusterAutoscalerState{}
+	diags := request.State.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resource := r.collection.Cluster(state.Cluster.Value).Autoscaler()
+	_, err := resource.Delete().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed deleting cluster autoscaler",
+			fmt.Sprintf(
+				"Failed deleting autoscaler for cluster '%s': %v",
+				state.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	response.State.RemoveResource(ctx)
+}
+
+func (r *ClusterAutoscalerResource) ImportState(ctx context.Context, request tfsdk.ImportResourceStateRequest,
+	response *tfsdk.ImportResourceStateResponse) {
+	clusterId := request.ID
+
+	getResponse, err := r.collection.Cluster(clusterId).Autoscaler().Get().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed importing cluster autoscaler",
+			err.Error(),
+		)
+		return
+	}
+
+	state := &ClusterAutoscalerState{}
+	populateAutoscalerState(getResponse.Body(), clusterId, state)
+	diags := response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
+
+func (r *ClusterAutoscalerResource) waitForClusterToGetReady(ctx context.Context, clusterId string) error {
+	resource := r.collection.Cluster(clusterId)
+
+	pollCtx, cancel := context.WithTimeout(ctx, 1*time.Hour)
+	defer cancel()
+
+	_, err := resource.Poll().
+		Interval(30 * time.Second).
+		Predicate(func(get *cmv1.ClusterGetResponse) bool {
+			return get.Body().State() == cmv1.ClusterStateReady
+		}).
+		StartContext(pollCtx)
+
+	return err
+}
+
+// populateAutoscalerState copies the data from the API object to the Terraform state.
+func populateAutoscalerState(object *cmv1.ClusterAutoscaler, clusterId string, state *ClusterAutoscalerState) {
+	state.Cluster = types.String{
+		Value: clusterId,
+	}
+
+	if value, exists := object.GetBalanceSimilarNodeGroups(); exists {
+		state.BalanceSimilarNodeGroups = types.Bool{Value: value}
+	} else {
+		state.BalanceSimilarNodeGroups = types.Bool{Null: true}
+	}
+
+	if value, exists := object.GetSkipNodesWithLocalStorage(); exists {
+		state.SkipNodesWithLocalStorage = types.Bool{Value: value}
+	} else {
+		state.SkipNodesWithLocalStorage = types.Bool{Null: true}
+	}
+
```

Comment:
can you export it to a common function? 

```suggestion
       state.SkipNodesWithLocalStorage =common.setBoolOptionalValue(object.GetSkipNodesWithLocalStorage())
       ......
func setBoolOptionalValue(value *bool, exists bool) types.Bool{      
       if exists {
              return types.Bool{Value: value}
       }
       return types.Bool{Null: true}
```